php/installer/Installer.class.inc

```diff
@@ -297,12 +297,16 @@ class Installer
     {
         $DB = Database::singleton();
         try {
-            $DB->connect(
+            $success = $DB->connect(
                 $values['dbname'],
                 $values['dbadminuser'],
                 $values['dbadminpassword'],
                 $values['dbhost']
             );
+
+            if (! $success) {
+                throw new \DatabaseException('Could not connect to database');
+            }
```

Comment:
looking at the code, it looks to me like we should be editing what's in the catch block below. looks to me like line 308 here is redundant to lines 314-315. maybe the solution here is to do something with `$this->_lastErr` e.g. displaying it on the front-end and/or throw the DatabaseException there

Reply:
ahhh i see